js/ui/search.js

```diff
@@ -367,14 +394,35 @@ const GridSearchResults = new Lang.Class({
 });
 Signals.addSignalMethods(GridSearchResults.prototype);
 
+const SearchResultsBin = new Lang.Class({
```

Comment:
Note that we still need to use our own subclass of St.BoxLayout to override the allocate vfunc, since max-width is not a CSS property that clutter understands out-of-the-box, but something that's implemented at the level of the Shell Toolkit:
https://developer.gnome.org/st/stable/st-st-theme-node.html#st-theme-node-get-max-width